packages/plugin-encryption/src/encryption.js

```diff
@@ -142,9 +142,27 @@ const Encryption = SparkPlugin.extend({
     }
 
     return this.unboundedStorage.get(uri)
+      .then((keyString) => JSON.parse(keyString))
+      .then((keyObject) => this.kms.asKey(keyObject))
       .catch(() => this.kms.fetchKey({uri})
-        .then((key) => this.unboundedStorage.put(key.uri, key)));
+        .then(tap((key) => this.unboundedStorage.put(key.uri, JSON.stringify(key, replacer)))));
   }
 });
 
+/**
+ * JSON.stringify replacer that ensures private key data is serialized.
+ * @param {string} k
+ * @param {mixed} v
+ * @returns {mixed}
+ */
+function replacer(k, v) {
```

Comment:
it ensures that when JSON.stringify is called on an encryption key, its toJSON method receives `true` so that the keys private data is included in the string